e2e/tests-telemetry/canister.bash

```diff
@@ -0,0 +1,188 @@
+
+    # events within 30 days go into the same reporting period
+    setTime 20
+
+    reportCommandSuccess 0.12.1 localShared darwin 'dfxCanisterCall'
+    reportCommandFailure 0.12.1 localShared darwin 'dfxCanisterCall'
+
+    assert_command getCommandSuccessRates "$(timestamp 0)" 0.12.1 localProject linux
+    assert_contains 'successRate = 75 : nat;'
+
+    assert_command getCommandSuccessRates "$(timestamp 0)" 0.12.1 localShared darwin
+    assert_contains 'successRate = 50 : nat;'
+
+    # on the start of the 30th day, events go into a new reporting period
+    setTime 30
+    reportCommandSuccess 0.12.1 localShared darwin 'dfxCanisterCall'
+    assert_command getCommandSuccessRates "$(timestamp 0)" 0.12.1 localShared darwin
+    assert_contains 'successRate = 50 : nat;'
+    assert_command getCommandSuccessRates "$(timestamp 30)" 0.12.1 localShared darwin
+    assert_contains 'successRate = 100 : nat;'
+}
+
+
+@test "stores entries by reporting period" {
+    dfx_start
+    dfx deploy
+
+    # first reporting period
+    setTime 0 23 # reporting period starts at the beginning of the day of the first event received
+    reportCommandSuccess 0.12.1 localShared linux 'dfxStart'
+    reportCommandSuccess 0.12.1 localShared linux 'dfxDeploy'
+    reportCommandFailure 0.12.1 localShared linux 'dfxStart'
+    reportCommandSuccess 0.12.1 localShared linux 'dfxStop'
+
+    setTime 7 2
+    reportCommandFailure 0.12.1 localShared linux 'dfxCanisterCall'
+    reportCommandSuccess 0.12.1 localShared linux 'dfxStart'
+    reportCommandSuccess 0.12.1 localShared linux 'dfxCanisterCall'
+
+    # second reporting period
+    setTime 30
+    reportCommandSuccess 0.12.1 localShared darwin 'dfxDeploy'
+    setTime 52 4
+    reportCommandSuccess 0.12.1 localShared darwin 'dfxCanisterCall'
+
+    # third reporting period
+    setTime 61
+    reportCommandSuccess 0.12.1 localProject linux 'dfxStart'
+    setTime 68
+    reportCommandSuccess 0.12.1 localProject linux 'dfxDeploy'
+    reportCommandFailure 0.12.1 localProject linux 'dfxDeploy'
+
+    assert_command getCommandResultReportingPeriods
+    assert_contains '1_668_556_800_000_000_000'
+    assert_contains '1_671_148_800_000_000_000'
+    assert_contains '1_673_740_800_000_000_000'
+
+    assert_command getInvocationDetailsForReportingPeriod 0
+    assert_eq '(
+  vec {
+    record {
+      dfxVersion = "0.12.1";
+      network = variant { localShared };
+      platform = variant { linux };
+    };
+  },
+)'
+
+    assert_command getInvocationDetailsForReportingPeriod 30
+    assert_eq '(
+  vec {
+    record {
+      dfxVersion = "0.12.1";
+      network = variant { localShared };
+      platform = variant { darwin };
+    };
+  },
+)'
+
+    assert_command getInvocationDetailsForReportingPeriod 60
+    assert_eq '(
+  vec {
+    record {
+      dfxVersion = "0.12.1";
+      network = variant { localProject };
+      platform = variant { linux };
+    };
+  },
+)'
```

Comment:
I don't see anything that tests where `getInvocationDetailsForReportingPeriod` produces a vec with length >1, but I don't see why it wouldn't work.

Reply:
Oh thanks, I meant to include that